workflows/cloudify_system_workflows/snapshots/credentials.py

```diff
@@ -69,99 +75,67 @@ def _get_hosts():
                 for node in wctx.nodes
                 if is_compute(node)]
 
-    def _dump_agent_key(self, node_id, agent_key_path, archive_cred_path):
+    def _dump_agent_key(self, node_id, agent_key, archive_cred_path):
         """Copy an agent key from its location on the manager to the snapshot
         dump
         """
         os.makedirs(os.path.join(archive_cred_path, node_id))
-        source = os.path.expanduser(agent_key_path)
+        source = os.path.expanduser(agent_key)
         destination = os.path.join(archive_cred_path, node_id,
                                    self._CRED_KEY_NAME)
         ctx.logger.debug('Dumping credentials data, '
                          'copy from: {0} to {1}'
                          .format(source, destination))
-        shutil.copy(source, destination)
-
-    def _restore_agent_credentials(self,
-                                   dep_node_id,
-                                   dump_cred_dir,
-                                   restored_cred_dir,
-                                   agent_key_path_dict):
-        restored_agent_key_path = \
-            self._restore_agent_key_from_dump(dump_cred_dir,
-                                              restored_cred_dir,
-                                              dep_node_id)
-        db_agent_key_path = agent_key_path_dict[dep_node_id]
-
-        if os.path.isfile(db_agent_key_path):
-            self._handle_existing_agent_key_path(
-                restored_agent_key_path,
-                db_agent_key_path
-            )
-        else:
-            copy_file(restored_agent_key_path, db_agent_key_path)
+        try:
+            shutil.copy(source, destination)
```

Comment:
Before copying, make sure this is a file with something like:
if os.path.exists(source) and os.path.isfile(source):